src/actions/profile-view.js

```diff
@@ -1171,6 +1171,28 @@ export function showGlobalTrack(trackIndex: TrackIndex): ThunkAction<void> {
   };
 }
 
+/**
+ * This action shows a specific global track and its local tracks.
+ */
+export function showGlobalTrackIncludingLocalTracks(
+  trackIndex: TrackIndex,
+  pid: Pid
```

Comment:
Curious we need both the trackIndex and pid. Isn't one of them enough?

Edit: I see it now that we use it for directly making the local tracks visible with `hiddenLocalTracksByPid.set(action.pid, new Set());` below. Maybe we can still get this information from the track reference, but probably not worth the effort that much.

Reply:
The main problem is that in reducers we don't have access to all the state. We need to provide in actions all data needed to handle actions in reducers. That's different to what happens in thunk actions where we have access to all the state. But I believe it's usually a better model, because the state is more self-contained this way, than in thunk actions where everything can interact with everything.